doc/pmon/pmon-sensormon.md

```diff
@@ -0,0 +1,145 @@
+
+Note that temperature sensor devices are managed via SONiC ThermalCtlD daemon today. At this point there is no change proposed for ThermalCtlD. This proposed design can be used for voltage, current and other types of sensors.
+
+
+### Requirements
+
+This HLD covers
+
+* Discovery of voltage and current sensor devices in the system
+* Monitoring the sensor devices periodically and update that data in Redis DB
+* Raise/Clear alarms if the sensor devices indicate readings which are unexpected and clear them if they return to a good state.
+* A framework for adding new sensor types in future
+
+This HLD does not cover
+
+* An automated recovery action that system might take as a result of a fault reported by the voltage sensor device. A network management system may process the alarms and take recvoery action as it sees fit.
+
+
+### High Level Design 
+
+The proposal for monitoring sensor devices is to create a new SensorMon daemon. SensorMon will use API provided by platform to discover the sensor devices. It will periodically poll the devices to refresh the sensor information and update the readings in StateDB.
+
+If the sensor device readings cross the minor/major/critical thresholds, syslogs will be generated to indicate to the operator about the alarm condition. If the sensor reports normal data in a subsequent poll, another syslog will be generated to indicate that the fault condition is cleared.
+
+CLIs are provided to display the sensor devices, their measurements, threshold values and if they are reporting an alarm.
+
+Platform APIs will provide 
+
+* List of sensors devices of a specific type in the system
+* Way to read the sensor information
+
+The following SONiC repositories will have changes
+
+#### sonic-platform-daemons	
+
+SensorMon will be a new daemon that will run in PMON container. It will retrieve a list of sensors of different sensor types from the platform during initialization. Subsequently, it will poll the sensor devices on a periodic basis and update their measurments in StateDb. SemsorMon will also raise syslogs on alarm conditions. 
```

Comment:
@bmridul sensord already does all these including raising ALERT
```
Nov  3 19:03:00.468898 sonic ALERT pmon#sensord: Sensor alarm: Chip dps460-i2c-10-59: PSU-1(L) 220V Rail (in): -589824.00 V (min = -0.50 V, max = -0.50 V) [ALARM]
Nov  3 19:03:00.469262 sonic ALERT pmon#sensord: Sensor alarm: Chip dps460-i2c-10-59: vout1: +536608768.00 V (min = +1071628288.00 V, max = +1071628288.00 V) [ALARM]
Nov  3 19:03:00.469388 sonic ALERT pmon#sensord: Sensor alarm: Chip dps460-i2c-10-59: PSU-1(L) Fan 1: -1 RPM [ALARM]
```

Reply:
Sensord does raise alarm if the device reports an alarm.
https://stuff.mit.edu/afs/sipb/user/dmaze/debian/Packages/lm-sensors/lm-sensors-2.8.0/doc/lm_sensors-FAQ.html#Section%204.4
However certain hardware designs will recommend monitoring voltage readings to generate alarms (even though device may not be reporting or not be capable of reporting an alarm).

Reply:
@bmridul can we specify customer threshold in sensors conf file

https://www.kernel.org/doc/Documentation/hwmon/sysfs-interface


<img width="718" alt="image" src="https://github.com/sonic-net/SONiC/assets/45705344/c114ac2d-a469-45d6-a287-b7ff02843100">

Reply:
<img width="425" alt="image" src="https://github.com/sonic-net/SONiC/assets/45705344/73cdc4bc-d18e-4cc7-9adb-b9919f275d21">